src/ExploreCharts.js

```diff
@@ -18,57 +18,100 @@ export default class ExploreCharts extends PureComponent {
     this.state = {
       departments: [],
       departmentId: null,
-      attribute: ETHNICITY_ATTRIBUTE
+      attribute: ETHNICITY_ATTRIBUTE,
+      dates: [],
+      date: null
     }
 
     this.setDepartmentId = this.setDepartmentId.bind(this)
     this.setAttribute = this.setAttribute.bind(this)
+    this.setDate = this.setDate.bind(this)
   }
 
-  loadDepartments () {
-    window.fetch('./data/departments.json')
+  fetchDepartments () {
+    return window.fetch('./data/departments.json')
       .then(res => res.json())
-      .then(departments => {
-        this.setState({departments})
-      })
-      .catch(console.error)
   }
 
-  loadEmployees () {
-    window.fetch('./data/employees.csv')
-    .then(res => res.text())
-    .then(text => {
+  fetchEmployees (date) {
+    return window.fetch(
+      `./data/${date}/employees.csv`
+    ).then(
+      res => res.text()
+    ).then(text => {
       const employees = csvParse(text)
       employees.forEach(employee => {
         employee.ethnicity = ETHNICITY_ID_LABELS[employee.ethnicityId]
       })
-      this.setState({employees})
-    }).catch(console.error)
+      return employees
+    })
+  }
+
+  fetchDates () {
+    return window.fetch(`./data/dates.json`)
+    .then(res => res.json())
   }
 
   componentDidMount () {
-    this.loadDepartments()
-    this.loadEmployees()
+    const datesPromise = this.fetchDates()
+    const employeesPromise = datesPromise.then(dates => {
+      const latestDate = dates[dates.length - 1]
+      return this.fetchEmployees(latestDate)
+    })
+    const departmentsPromise = this.fetchDepartments()
+
+    Promise.all(
+      [datesPromise, departmentsPromise, employeesPromise]
+    ).then(results => {
+      const [dates, departments, employees] = results
+      this.setState({
+        dates: dates,
+        date: dates[dates.length - 1],
+        departments,
+        employees}
+      )
+    }).catch(console.error)
   }
 
   render () {
     const departmentOptions = Object.entries(this.state.departments).map(
       ([id, name]) => ({ label: name, value: id })
     )
 
+    const dateOptions = this.state.dates.map(
+      value => ({ label: value, value: value })
+    )
+
     return (
       <section className='ExploreCharts'>
         <div className='ExploreCharts__SelectContainer'>
-          <h2>Select a department and a metric to generate a custom report for the most recent year</h2>
+          <label for='explore-charts-date-select'>Date</label>
```

Comment:
`for` should be `htmlFor`
https://facebook.github.io/react/docs/dom-elements.html#htmlfor

Reply:
![ming blown](http://www.reactiongifs.com/wp-content/uploads/2011/09/mind_blown.gif)

Thanks!